client/components/domains/register-domain-step/index.jsx

```diff
@@ -337,9 +337,26 @@ class RegisterDomainStep extends Component {
 		return strippedHostname ?? hostname;
 	}
 
+	getInitialQueryInAIFlow() {
+		if ( this.props.flowName !== 'ai-site-builder' ) {
+			return;
+		}
+
+		// TODO: Maybe reading from meta or url would be useful too.
+		if ( ! this.props.selectedSite || ! this.props.selectedSite.name ) {
+			return;
+		}
+
+		return this.props.selectedSite.name;
+	}
+
 	componentDidMount() {
 		const storedQuery = globalThis?.sessionStorage?.getItem( SESSION_STORAGE_QUERY_KEY );
-		const query = this.state.lastQuery || storedQuery || this.getInitialQueryInLaunchFlow();
+		const query =
+			this.state.lastQuery ||
+			storedQuery ||
+			this.getInitialQueryInAIFlow() ||
+			this.getInitialQueryInLaunchFlow();
```

Comment:
See, if this was generic, you would have just used it.

Reply:
Is that better?
https://github.com/Automattic/wp-calypso/pull/100948/commits/82ef7eb326213c02f38296ab314a690ba9e8b7cf